Tests/scripts/destroy_instances.py

```diff
@@ -2,81 +2,126 @@
+                     f"({humanize.naturalsize(sent, binary=True, gnu=True)}/{humanize.naturalsize(size, binary=True, gnu=True)})")
 
-    except subprocess.CalledProcessError:
-        logging.exception(f'Failed to shutdown server {server_ip}')
+    try:
+        download_path = (Path(artifacts_dir) / f"server_{role}_{server_ip}.log").as_posix()
+        logging.info(f'Downloading server logs from server {server_ip} from:{SERVER_LOG_FILE_PATH} to {download_path}')
+        with SCPClient(ssh.get_transport(), progress4=progress) as scp:
+            scp.get(SERVER_LOG_FILE_PATH, download_path)
+        return True
+    except SCPException:
+        logging.exception(f'Failed downloading server logs from server {server_ip}')
+    return False
 
 
 def main():
     install_logging('Destroy_instances.log')
     options = options_handler()
-    time_to_live = os.getenv('TIME_TO_LIVE')
+    time_to_live = int(os.getenv('TIME_TO_LIVE') or DEFAULT_TTL)
     tests_path = Path('./Tests')
+    start_time = datetime.utcnow()
+    logging.info(f"Starting destroy instances - environment from {options.env_file}, TTL:{time_to_live} seconds, "
+                 f"Tests Path:{tests_path.absolute()}")
 
-    with open(options.env_file, 'r') as json_file:
+    with open(options.env_file) as json_file:
         env_results = json.load(json_file)
 
-    for env in filter(lambda x: x["Role"] == options.instance_role, env_results):
+    servers_list = list(filter(lambda x: x["Role"] == options.instance_role, env_results))
+    logging.info(f"Found {len(servers_list)} server from the environment file to destroy")
+    success = True
+    for i, env in enumerate(servers_list, 1):
         readable_role = env["Role"]
         role = readable_role.replace(' ', '')
         server_ip = env["InstanceDNS"]
-
-        logging.info(f'Downloading server log from {readable_role}')
-        chmod_logs(server_ip)
-        download_logs(server_ip, options.aritfacts_dir, role)
-
-    if time_to_live:
-        logging.info(f'Time to live was set to {time_to_live} minutes')
-        shutdown(server_ip, int(time_to_live))
-    elif (tests_path / f'is_build_passed_{role}.txt').exists() and \
-            (tests_path / f'is_post_update_passed_{role}.txt').exists():
-        shutdown(server_ip)
-    else:
-        logging.warning(f'Tests for some integration failed on {readable_role}, keeping instance alive')
+        logging.info(f'{i}/{len(servers_list)} - Downloading server log from {readable_role}, ip:{server_ip} and destroying it')
+
+        with SSHClient() as ssh:
+            try:
+                ssh.load_system_host_keys()
+                ssh.connect(server_ip, username=SSH_USER)
+                success &= chmod_logs(ssh, server_ip)
+                success &= download_logs(ssh, server_ip, options.artifacts_dir, role)
+
+                if time_to_live:
+                    logging.info(f'Time to live was set to {time_to_live} minutes for server:{server_ip}')
```

Comment:
Minutes, or seconds?
```suggestion
                    logging.info(f'Time to live was set to {time_to_live} minutes for server {server_ip}')
```